jib-core/src/main/java/com/google/cloud/tools/jib/configuration/credentials/CredentialRetriever.java

```diff
@@ -25,9 +25,14 @@
   /**
    * Fetches the credentials. <b>Implementations must be thread-safe.</b>
    *
+   * <p>Implementations should return {@code null} if no credentials could be fetched with this
+   * {@link CredentialRetriever} (and so other credential retrieval methods may be tried), or throw
+   * an exception something went wrong when fetching the credentials.
+   *
    * @return the fetched credentials or {@code null} if no credentials could be fetched with this
    *     provider
+   * @throws Exception if the credential retrieval encountered an exception
    */
   @Nullable
-  Credentials retrieve();
+  Credential retrieve() throws Exception;
```

Comment:
I agree with Brian's comment about using a more specific exception.

Reply:
Added that to the TODOs of #832